### PR TITLE
Improvement/cli usability

### PR DESCRIPTION
**Pull request by EarlPitts**
Extending the CLI to support more options.

### CODE REVIEW

**Comment by EarlPitts** [comment]
A bit of context:
The main change in this PR is to checkout the two different versions beforehand, and use these folders for the comparison.
On the one hand this avoids the need to repeatedly check out the different versions while evaluating the functions, but more importantly, we can specify on the command line whether we want to compare a commit or a folder (and any combination of those two).

File: src/diff.erl [hunk 1]
```diff
@@ -62,20 +62,20 @@ modified_files(Diffs) ->
...
-% Extracts the filename from the diff output
+% Extracts the filenames from the diff output
 -spec extract_file(string()) -> filename().
 extract_file(DiffLine) ->
-    Options = [global, {capture, [1], list}],
-    {match, [[FileName]]} = re:run(DiffLine,".*a/(.*?\.erl).*", Options),
-    FileName.
+    Options = [global, {capture, [1,2], list}],
+    {match, [[OrigFile, RefacFile]]} = re:run(DiffLine, ".*?(/.*?\.erl).*?(/.*?\.erl)", Options),
+    tl(lists:dropwhile(fun(C) -> C =/= $/ end, utils:common_postfix(OrigFile, RefacFile))). % This is too brittle
```
Comment (daniel-horpacsi) [inline src/diff.erl:78]: Make it less brittle before merging to master.

File: src/cli.erl [hunk 2]
```diff
@@ -0,0 +1,104 @@
+-module(cli).
+
+-include("equivchecker.hrl").
+
+-export([run/1]).
+
+% TODO Refactor this monstrosity into something manageable
+-spec handler(map()) -> none().
+handler(#{target := Target, source := Source, json := Json, commit := Commit, stats := Stats}) when Commit ->
+    if not Json -> io:format("Checking commit ~p against commit ~p~n", [Target, Source]); true -> ok end,
```
Comment (daniel-horpacsi) [inline src/cli.erl:10]: Refactor to `Json andalso io:format`?

File: src/cli.erl [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    % Creating temporary folders for the source and bytecode
+    file:make_dir(?TEMP_FOLDER),
+    file:make_dir(?ORIGINAL_BIN_FOLDER),
+    file:make_dir(?REFACTORED_BIN_FOLDER),
+
+    application:start(wrangler).
+
+cleanup() ->
+    ets:delete(stat),
+    file:del_dir_r(?TEMP_FOLDER), % TODO Handle error
```
Comment (daniel-horpacsi) [inline src/cli.erl:60]: Or just put it under a `catch`.

File: src/cli.erl [hunk 4]
```diff
@@ -0,0 +1,104 @@
+-module(cli).
+
+-include("equivchecker.hrl").
+
+-export([run/1]).
+
+% TODO Refactor this monstrosity into something manageable
```
Comment (daniel-horpacsi) [inline src/cli.erl:7]: I agree. Also, get rid of code duplication where possible.

File: src/cli.erl (hunk 3 above)
Comment (EarlPitts) [reply src/cli.erl:60]: My plan was to deal with these kind of errors in the battle testing phase.
And I'm not even sure this should be checked explicitly.
Handling the case when the creation if this temp folder fails would make more sense.

File: src/diff.erl (hunk 1 above)
Comment (EarlPitts) [reply src/diff.erl:78]: This is tangentially related to  #15.
I've already started working on another branch to fix it, and that will probably change how this works too, so I will deal with this there.

File: src/cli.erl (hunk 2 above)
Comment (EarlPitts) [reply src/cli.erl]: Thanks, I've updated it.

File: include/equivchecker.hrl [hunk 5]
```diff
@@ -1,3 +1,11 @@
+-define(TEMP_FOLDER, "tmp"). % TODO use /tmp
+
+-define(ORIGINAL_BIN_FOLDER, ?TEMP_FOLDER ++ "/orig").
```
[the remark this answers is not in the document]
Comment (EarlPitts) [reply include/equivchecker.hrl]: Done.

File: src/cli.erl (hunk 3 above)
Comment (daniel-horpacsi) [reply src/cli.erl:79]: Alright. I'd suggest that you don't leave these TODO comments hanging around in PRs but turn them into proper issues/feature requests before the merge.

**Pull request merged**